components/merkledb/src/proof_list_index/proof.rs

```diff
@@ -311,3 +298,58 @@ where
         Ok(res)
     }
 }
+
+impl<'a, V> Deserialize<'a> for ListProof<V>
```

Comment:
Having to write such crutches is one of the reasons I'd prefer non-recursive `ListProof` definition. I've written [an implementation](https://github.com/slowli/exonum/tree/list-proof-this-time-its-personal) (although it's severely outdated), and I'll draft the corresponding internal RFC soon.